pkg/cloudprovider/providers/openstack/openstack_instances.go

```diff
@@ -63,6 +71,11 @@ func (os *OpenStack) Instances() (cloudprovider.Instances, bool) {
 // CurrentNodeName implements Instances.CurrentNodeName
 // Note this is *not* necessarily the same as hostname.
 func (i *Instances) CurrentNodeName(ctx context.Context, hostname string) (types.NodeName, error) {
+	// if compute is not initialized just return hostname
+	if i.compute == nil {
+		return types.NodeName(hostname), nil
```

Comment:
So is this local hostname override only required for kubelet TLS bootstrapping?

Reply:
yes, which is consistent with how this will work w/ external cloud provider.

Reply:
What do you think of https://github.com/kubernetes/kubernetes/pull/75229 where we try to discover the hostname and internal IPs if it wasn't already set by the external cloud provider. That would fix this problem for all cloud providers not just OpenStack.

Reply:
Yes, we override hostnames only for kubelet TLS bootstrapping if the config is taken from a secret.
There's an idea that for consistency, we should always return the hostname. What do you think?

Reply:
I believe with https://github.com/kubernetes/kubernetes/pull/75229 we would return the hostname for all providers using the external providers.

Reply:
Yeah, #75229 makes sense to me and solves the issue for external cloud providers.
Here we are trying to do the same for in-tree OpenStack cloud-provider, which is the only one that gives out different names for hosts and nodes. 
So I think these patches complement each other.

Reply:
I see, so I had the impression that this was mainly to help the transition to external providers since not setting hostname breaks TLS bootstrapping. So is TLS bootstrapping also broken for the in-tree case?